wordpress-mobile/WordPress-iOS
# Categories: Add refresh control and sync for categories.

Fixes #2573 

This adds a simple `UIRefreshControl` to the `PostCategoriesViewController` and reloads the table once synced. This resolves the issue of users not being able to pull in any recent changes to categories if the app hasn't synced the blog. It also wasn't obvious how to refresh categories without force-closing the app and relaunching.

To test:
1. Add a self-hosted site.
2. Go straight to Post Options > Categories.
3. Check that the list is currently in sync with the web.
4. Make a change on the web.
5. Pull down the refresh, check that the list is again in sync with the web.

To test:
1. Repeat steps for a .com or Jetpack site.

Needs review: @frosty can you give this is a gander?
